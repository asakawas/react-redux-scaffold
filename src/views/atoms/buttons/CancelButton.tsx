import React, { FunctionComponent } from 'react';
import SpacingButton, { SpacingButtonProps } from '~/views/atoms/buttons/SpacingButton';

/**
 * The type of props of CancelButton.
 */
export type CancelButtonProps = Readonly<
  Pick<SpacingButtonProps, 'ml'> & {
    /** The event handler called when the button is clicked */
    onCancelButtonClick: SpacingButtonProps['onClick'];
  }
>;

const CancelButton: FunctionComponent<CancelButtonProps> = ({ onCancelButtonClick, ml }) => (
  <SpacingButton
    variant="contained"
    aria-label="cancel button"
    onClick={onCancelButtonClick}
    ml={ml}
  >
    Cancel
  </SpacingButton>
);

export default CancelButton;
