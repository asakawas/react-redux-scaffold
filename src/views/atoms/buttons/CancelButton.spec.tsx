import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import CancelButton from './CancelButton';

describe('CancelButton', () => {
  test('is rendered as expected', () => {
    const { asFragment } = render(<CancelButton onCancelButtonClick={() => {}} ml={1} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("calls the passed handler when it's clicked", () => {
    const handler = jest.fn();
    const renderResult = render(<CancelButton onCancelButtonClick={handler} ml={1} />);

    const tree = renderResult.getByText('Cancel');
    fireEvent.click(tree);
    expect(handler).toHaveBeenCalledTimes(1);

    fireEvent.click(tree);
    expect(handler).toHaveBeenCalledTimes(2);
  });

  test('has an attribute "ml" with its value set to the given number', () => {
    const renderResult = render(<CancelButton onCancelButtonClick={() => {}} ml={11} />);

    const btn = renderResult.getByLabelText('cancel button');
    expect(btn).toHaveAttribute('ml', '11');
  });
});
