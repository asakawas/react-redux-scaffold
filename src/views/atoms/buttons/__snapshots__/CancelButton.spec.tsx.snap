// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CancelButton is rendered as expected 1`] = `
<DocumentFragment>
  .c0 {
  margin-left: 8px;
}

<button
    aria-label="cancel button"
    class="MuiButtonBase-root MuiButton-root MuiButton-contained c0"
    ml="1"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Cancel
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
